nodeup/pkg/model/context.go

```diff
@@ -329,8 +329,8 @@ func (c *NodeupModelContext) UseChallengeCallback(cloudProvider kops.CloudProvid
 	return model.UseChallengeCallback(cloudProvider)
 }
 
-func (c *NodeupModelContext) UseExternalECRCredentialsProvider() bool {
-	return model.UseExternalECRCredentialsProvider(c.kubernetesVersion, c.CloudProvider())
+func (c *NodeupModelContext) UseExternalKubeletCredentialProvider() (bool, kops.CloudProviderID) {
```

Comment:
Why is returning `kops.CloudProviderID` neede? Shouldn't it be part of `NodeupModelContext`?

Reply:
We want the function to behave differently depending on the cloud provider version that is passed through(look at the diff in features.go)

Reply:
See my latest suggestions

Reply:
Fixed

Reply:
Now there is no need for the new return value (not used anywhere).